dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/AsyncRpcResult.java

```diff
@@ -144,7 +148,12 @@ public Result getAppResponse() {
             // This should never happen;
             logger.error("Got exception when trying to fetch the underlying result from AsyncRpcResult.", e);
         }
-        return new AppResponse();
+
+        ConsumerModel consumerModel = ApplicationModel.getConsumerModel(invocation.getTargetServiceUniqueName());
+        String methodName = invocation.getMethodName();
```

Comment:
I think you could get  consumerModel directly from invocation attributes

Reply:
done.